cmake-modules/AddGoogleTest.cmake

```diff
@@ -0,0 +1,98 @@
+# 
+#
+# Downloads GTest and provides a helper macro to add tests. Add make check, as well, which
+# gives output on failed tests without having to set an environment variable.
+#
+#
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+
+if(CMAKE_VERSION VERSION_LESS 3.11)
+    set(UPDATE_DISCONNECTED_IF_AVAILABLE "UPDATE_DISCONNECTED 1")
+
+    include(DownloadProject)
+    download_project(PROJ                googletest
+		     GIT_REPOSITORY      https://github.com/google/googletest.git
+		     GIT_TAG             master
+		     UPDATE_DISCONNECTED 1
+		     QUIET
+    )
+
+    # CMake warning suppression will not be needed in version 1.9
+    set(CMAKE_SUPPRESS_DEVELOPER_WARNINGS 1 CACHE BOOL "")
+    add_subdirectory(${googletest_SOURCE_DIR} ${googletest_SOURCE_DIR} EXCLUDE_FROM_ALL)
+    unset(CMAKE_SUPPRESS_DEVELOPER_WARNINGS)
+else()
+    include(FetchContent)
+    FetchContent_Declare(googletest
+        GIT_REPOSITORY      https://github.com/google/googletest.git
+        GIT_TAG             master)
```

Comment:
there is a weird behavior if selecting a TAG like release-1.8.0
I might need to ask Charlie about it. But basically I can't compile any other but master.
Any tag throws hundreds of warnings